search/includes/classes/class-search.php

```diff
@@ -1485,8 +1547,12 @@ function( $value ) {
 		return $post_meta_allow_list;
 	}
 
+	public function get_post_meta_allow_list( $post ) {
+		return $this->parse_vip_search_post_meta_allow_list_filter( $post );
+	}
+
 	public function filter__ep_prepare_meta_allowed_protected_keys( $keys, $post ) {
-		return \apply_filters( 'vip_search_post_meta_allow_list', $keys, $post );
+		return $this->parse_vip_search_post_meta_allow_list_filter( $post, $keys );
```

Comment:
The previous implementation would fail if the vip_search_post_meta_allow_list would return assoc_array here, so I lifted up that logic to a new function `parse_vip_search_post_meta_allow_list_filter`